### PR TITLE
Upgrade fixes and improvements

### PR DESCRIPTION
**Pull request by vinnyjth**
## TODO:

- [ ] PR has a relevant title that will be understandable in a public changelog (ie...non developers)
- [ ] Closes [tag-issues-here]
- [ ] No new warnings in tests, in storybook, and in-app
- [ ] Upload GIF(s) of iOS and Android
- [ ] Set a relevant reviewer

## REVIEW:

**Manual QA**

- [ ] Manual QA on iOS and ensure it looks/behaves as expected
- [ ] Manual QA on Android and ensure it looks/behaves as expected

**Code Review: Questions to consider**

- [ ] Read through the "Files changed" tab _very carefully_
- [ ] Edge cases: what assumptions are made about input?
- [ ] What kind of tests could be written?
- [ ] How might we make this easier for someone else to understand?
- [ ] Could the code be simpler?
- [ ] Will the code be easy to modify in the future?
- [ ] What's one part of these changes that makes you excited to merge it?

> The purpose of PR Review is to _improve the quality of the software._

### CODE REVIEW

**Comment by vinnyjth** [comment]
@conrad-vanl can you give this a test for me? It fixes the "Hello null" bug and ensures that nobody will be matched to their children if their children are the only ones with a phone number.

File: packages/apollos-api/src/data/auth/index.js [hunk 1]
```diff
@@ -3,14 +3,13 @@ import { Auth } from '@apollosproject/data-connector-rock';
...
-  createUserProfile = async (props = {}) => {
+  createUserProfile = async ({ email, ...otherFields }) => {
     try {
-      const { email } = props;
-
       return await this.post('/People', {
         Email: email,
-        IsSystem: false, // Required by Rock
         Gender: 0, // Required by Rock
+        ...otherFields,
```
Comment (conrad-vanl) [inline packages/apollos-api/src/data/auth/index.js]: @vinnyjth should this not `PascalCase` the keys? Won't they be `camcelCase`d by default?

Comment (vinnyjth) [reply packages/apollos-api/src/data/auth/index.js]: The GQL inputs keys are all CamelCased, so we'll be safe. 

https://github.com/ApollosProject/apollos-prototype/blob/master/packages/apollos-church-api/local.graphql#L488

Comment (conrad-vanl) [reply packages/apollos-api/src/data/auth/index.js]: you mean `PascalCase`d, but yeah okay cool.
